lib/location/method_gnss.c

```diff
@@ -0,0 +1,835 @@
+				    resp_buf_cellid,
+				    MODEM_INFO_MAX_RESPONSE_SIZE);
+	if (err < 0) {
+		LOG_ERR("Could not obtain cell id");
+		return err;
+	}
+
+	err = modem_info_string_get(MODEM_INFO_AREA_CODE,
+				    resp_buf_tac,
+				    MODEM_INFO_MAX_RESPONSE_SIZE);
+	if (err < 0) {
+		LOG_ERR("Could not obtain tracking area code");
+		return err;
+	}
+
+	/* Request for MODEM_INFO_OPERATOR returns both MNC and MCC in the same string. */
+	err = modem_info_string_get(MODEM_INFO_OPERATOR,
+				    resp_buf_cops,
+				    MODEM_INFO_MAX_RESPONSE_SIZE);
+	if (err < 0) {
+		LOG_ERR("Could not obtain current operator information");
+		return err;
+	}
+
+	serving_cell->id = strtol(resp_buf_cellid, NULL, 16);
+	serving_cell->tac = strtol(resp_buf_tac, NULL, 16);
+	serving_cell->mnc = strtol(&resp_buf_cops[3], NULL, 10);
+	/* Null-terminated MCC, read and store it. */
+	resp_buf_cops[3] = '\0';
+	serving_cell->mcc = strtol(resp_buf_cops, NULL, 10);
+
+	return 0;
+}
+
+static void method_gnss_agps_request_work_fn(struct k_work *item)
+{
+	struct nrf_cloud_rest_context rest_ctx = {
+		.connect_socket = -1,
+		.keep_alive = false,
+		.timeout_ms = CONFIG_NRF_CLOUD_REST_RECV_TIMEOUT * MSEC_PER_SEC,
+		.auth = jwt_buf,
+		.rx_buf = rest_api_recv_buf,
+		.rx_buf_len = sizeof(rest_api_recv_buf),
+		.fragment_size = 0
+	};
+
+	int err = nrf_cloud_jwt_generate(0, jwt_buf, sizeof(jwt_buf));
+
+	if (err) {
+		LOG_ERR("Failed to generate JWT, error: %d", err);
+		return;
+	}
+
+	struct nrf_cloud_rest_agps_request request = {
+						      NRF_CLOUD_REST_AGPS_REQ_CUSTOM,
+						      &agps_request,
+						      NULL};
+	struct lte_lc_cells_info net_info = {0};
+
+	/* Get network info for the A-GPS location request. */
+	err = method_gnss_get_modem_info(&net_info.current_cell);
+
+	if (err) {
+		LOG_WRN("Requesting A-GPS data without location assistance");
+	} else {
+		request.net_info = &net_info;
+	}
+
+	struct nrf_cloud_rest_agps_result result = {agps_data_buf, sizeof(agps_data_buf), 0};
+
+	err = nrf_cloud_rest_agps_data_get(&rest_ctx, &request, &result);
+	if (err) {
+		LOG_ERR("nRF Cloud A-GPS request failed, error: %d", err);
+		return;
+	}
+
+	LOG_DBG("A-GPS data requested");
+
+	err = nrf_cloud_agps_process(result.buf, result.agps_sz);
+	if (err) {
+		LOG_ERR("A-GPS data processing failed, error: %d", err);
+		return;
+	}
+
+	LOG_DBG("A-GPS data processed");
+}
+#endif // defined(CONFIG_NRF_CLOUD_MQTT)
+#endif // defined(CONFIG_NRF_CLOUD_AGPS)
+
+#if defined(CONFIG_NRF_CLOUD_PGPS) && !defined(CONFIG_NRF_CLOUD_MQTT)
+static void method_gnss_pgps_request_work_fn(struct k_work *item)
+{
+	struct nrf_cloud_rest_context rest_ctx = {
+		.connect_socket = -1,
+		.keep_alive = false,
+		.timeout_ms = CONFIG_NRF_CLOUD_REST_RECV_TIMEOUT * MSEC_PER_SEC,
+		.auth = jwt_buf,
+		.rx_buf = rest_api_recv_buf,
+		.rx_buf_len = sizeof(rest_api_recv_buf),
+		.fragment_size = 0
+	};
+
+	int err = nrf_cloud_jwt_generate(0, jwt_buf, sizeof(jwt_buf));
```

Comment:
We need to check here also what to do with 0 time, meaning default.

Reply:
Closing, see comment https://github.com/nrfconnect/sdk-nrf/pull/5963#discussion_r740897508